docs/build_node.md

```diff
@@ -620,26 +620,51 @@ LDK will automatically reject incoming HTLCs to protect your privacy.
 all at once.
 
 ### Sending Payments
-**NOTE: CURRENTLY UNSUPPORTED IN JAVA**
 
 **Example:**
 ```java
-String invoice = // get an invoice from the payee
-Result_InvoiceNoneZ parsed_invoice = invoice.from_str();
+
+String invoice_str = // get an invoice from the payee
+Result_InvoiceNoneZ parsed_invoice = invoice_str.from_str();
+
 if (parsed_invoice instanceof Result_InvoiceNoneZ.Result_InvoiceNoneZ_OK) {
-	byte[] payment_hash = ((Result_InvoiceNoneZ.Result_InvoiceNoneZ_OK) parsed_invoice).res.payment_hash();
-	byte[] payment_secret = ((Result_InvoiceNoneZ.Result_InvoiceNoneZ_OK) parsed_invoice).res.payment_secret();
+	Invoice invoice = ((Result_InvoiceNoneZ.Result_InvoiceNoneZ_OK) parsed_invoice).res;
+    long amt = 0;
+    if (invoice.amount_pico_btc() instanceof Option_u64Z.Some) {
+        amt = ((Option_u64Z.Some)invoice.amount_pico_btc()).some;
+    }
+    if (amt == 0) {
+        // <Handle a zero-value invoice>
+    }
+
+    // LDK currently only supports one-hop route hints, so we need to select
+    // only the supported hints and convert to an array.
+    ArrayList<RouteHintHop> invoice_hops = new ArrayList<RouteHintHop>();
+    for (RouteHint hint : invoice.res.routes()) {
+            RouteHintHop[] hops = hint.into_inner();
+            if (hops.length == 1) {
+                    invoice_hops.add(hops[0]);
+            }
```

Comment:
Hm, so in the sample right now the equivalent code is this:
```
					let mut route_hints = invoice.routes().clone();
					let mut last_hops = Vec::new();
					for hint in route_hints.drain(..) {
						last_hops.push(hint[hint.len() - 1].clone());
					}
```

Meaning, we take the last hop from every `route`. Could we just do this instead of only using routes if they're 1-hop?

Reply:
Hmm, indeed you could! Its unlikely those hints are useful, but it would be correct to do so.